test/state/state.cpp

```diff
@@ -405,6 +435,54 @@ std::variant<TransactionReceipt, std::error_code> transition(State& state, const
     assert(sender_acc.nonce < Account::NonceMax);  // Checked in transaction validation
     ++sender_acc.nonce;                            // Bump sender nonce.
 
+    auto delegation_refund = 0;
+
+    for (const auto& auth : tx.authorization_list)
+    {
+        if (auth.chain_id != 0 && auth.chain_id != tx.chain_id)
+            continue;
+
+        // Check if signer could be ecrecovered.
+        if (!auth.signer.has_value())
+            continue;
+
+        // Check if authority exists.
+        auto* authority_ptr = state.find(*auth.signer);
+        if (authority_ptr != nullptr && !authority_ptr->is_empty())
+        {
+            authority_ptr->access_status = EVMC_ACCESS_WARM;
+
+            // Skip if authority has non-delegated code.
+            if (!authority_ptr->code.empty() && !is_code_delegated(authority_ptr->code))
+                continue;
+
+            // Skip if authorization nonce is incorrect.
+            if (auth.nonce != authority_ptr->nonce)
+                continue;
+
+            // Refund if authority account creation is not needed.
+            static constexpr int64_t EXISTING_AUTHORITY_REFUND = 25000 - 2500;
```

Comment:
Use constants here? we have `per_empty_account_cost`, need to add the other one.

Reply:
Made it two global constants